Documentation/capabilities-guide.md

```diff
@@ -0,0 +1,235 @@
+# Capabilities Isolators Guide
+
+This document is a walk-through guide describing how to use rkt isolators for
+[Linux Capabilities](https://lwn.net/Kernel/Index/#Capabilities).
+
+* [About Linux Capabilities](#about-linux-capabilities)
+* [Default Capabilities](#default-capabilities)
+* [Capability Isolators](#capability-isolators)
+* [Usage Example](#usage-example)
+* [Recommendations](#recommendations)
+
+## About Linux Capabilities
+
+Linux capabilities are meant to be a modern evolution of traditional UNIX
+permissions checks.
+The goal is to split the permissions granted to privileged processes into a set
+of capabilities (eg. `CAP_SYS_CHROOT` to perform a `chroot(2)`), which can be
+separately handled and assigned to single threads.
+
+Processes can gain specific capabilities by either being run by superuser, or by
+having the setuid/setgid bits or specific file-capabilities set on their
+executable file.
+Once running, each process has a bounding set of capabilities which it can
+enable and use; such process cannot get further capabilities outside of this set
+(except by using one of methods mentioned above).
+
+In the context of containers, capabilities are useful for:
+ * restricting the effective privileges of applications running as root
+ * allowing applications to perform specific privileged operations, without
+   having to run them as root
+
+For the complete list of existing Linux capabilities and a detailed description
+of this security mechanism, see [capabilities(7)](http://man7.org/linux/man-pages/man7/capabilities.7.html).
+
+## Default capabilities
+
+By default, rkt enforces [a default set of
+capabilities](https://github.com/appc/spec/blob/master/spec/ace.md#oslinuxcapabilities-remove-set)
+onto applications.
+This default set is tailored to stop applications from performing a large 
+variety of privileged actions, while not not impacting their normal behavior.
+Operations which are typically un-needed in containers and which may
+impact host state, eg. invoking `reboot(2)`, are denied in this way.
+
+However, this default set is mostly meant as a safety precaution against erratic
+and misbehaving applications, and will not suffice against tailored attacks.
+As such, it is recommended to fine-tune capabilities bounding set using one of 
+the customizable isolators available in rkt.
+
+## Capability Isolators
+
+When running Linux containers, rkt provides two mutually exclusive isolators
+to define the bounding set under which an application will be run:
+ * `os/linux/capabilities-retain-set`
+ * `os/linux/capabilities-remove-set`
+
+Those isolators cover different use-cases and employ different techniques to
```

Comment:
It serves as an intro to the whitelisting/blacklisting in each sub-chapter and to explain/remind the "mutual exclusive" part.